File: 215-kth-largest-element-in-an-array/kth-largest-element-in-an-array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        Queue<Integer> queue = new PriorityQueue<>((a,b) -> b - a);

        for(int num : nums){
            queue.add(num);
        }
        

        for(int i = 0; i < k -1;i++){
            queue.poll();
        }

        return queue.poll();
        
    }
}

// SC -> O(N)
// TC -> O(N Log K)

File: 215-kth-largest-element-in-an-array/kth-largest-element-in-an-array.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        min_heap = nums[:k] # [3,2]

        heapq.heapify(min_heap)

        for num in nums[k:]: # [1,5,6,4]
            if num > min_heap[0]:
                heapq.heapreplace(min_heap,num) # [5,6]
        
        return min_heap[0]
    
    # SC -> O(N)
    # TC -> O(N Log K)